File: src/main/java/com/duxtinto/intellij/plugin/github/codereviews/domain/pullrequests/GetAllOpenForRepoInteractor.java
package com.duxtinto.intellij.plugin.github.codereviews.domain.pullrequests;

import com.duxtinto.intellij.plugin.github.codereviews.domain.Interactor;
import com.duxtinto.intellij.plugin.github.codereviews.domain.repositories.GithubRepositoryEntity;

import java.io.IOException;
import java.util.List;

public class GetAllOpenForRepoInteractor implements Interactor<GithubRepositoryEntity, List<PullRequestEntity>> {
    private final PullRequestDomainContract.Repository repository;

    public GetAllOpenForRepoInteractor(PullRequestDomainContract.Repository repository) {
        this.repository = repository;
    }

    @Override
    public List<PullRequestEntity> run(GithubRepositoryEntity repo) {
        try {
            return repository.getAllOpenBy(repo.ownerName(), repo.name());
        } catch (IOException e) {
            e.printStackTrace();
            throw new RuntimeException(e);
        }
    }
}

File: src/main/kotlin/com/duxtinto/intellij/plugin/github/codereviews/domain/pullrequests/GetAllOpenForRepoInteractor.kt
package com.duxtinto.intellij.plugin.github.codereviews.domain.pullrequests

import com.duxtinto.intellij.plugin.github.codereviews.domain.repositories.GithubRepositoryEntity
import java.io.IOException
import javax.inject.Inject

class GetAllOpenForRepoInteractor
    @Inject
    constructor(private val repository: PullRequestDomainContract.Repository)
    : PullRequestsByRepositoryInteractor {

    override fun run(request: GithubRepositoryEntity): List<PullRequestEntity>? {
        try {
            return repository.getAllOpenBy(request.ownerName, request.name)
        } catch (e: IOException) {
            e.printStackTrace()
            throw RuntimeException(e)
        }

    }
}
